### PR TITLE
Feat: rabbit desire

### PR DESCRIPTION
**Pull request by fjebaker**

### CODE REVIEW

File: cart/rabbit.go [hunk 1]
```diff
@@ -30,14 +58,73 @@ func (rabbit *Rabbit) switchIdleFrame() {
 	rabbit.Sprite.Id = rabbit_main_frame + (4 * rabbit.Frame)
 }
 
+func (rabbit *Rabbit) drawThoughBubble(x, y, t int32) {
+	sprite := tic80.SquareSprite(rabbit.DesiredItem, 1)
+	x_item := x - RABBIT_ITEM_OFFSET_X
+	y_item := y - RABBIT_ITEM_OFFSET_X
```
Comment (JR-Mitchell) [inline cart/rabbit.go]: Should be `RABBIT_ITEM_OFFSET_Y`

File: cart/rabbit.go [hunk 2]
```diff
@@ -30,14 +58,73 @@ func (rabbit *Rabbit) switchIdleFrame() {
...
 	if t%45 == 0 {
 		rabbit.switchIdleFrame()
 	}
-	rabbit.Sprite.Draw(rabbit.X+PLAYER_OFFSET_X, rabbit.Y+PLAYER_OFFSET_Y)
+
+	x := rabbit.X + PLAYER_OFFSET_X
+	y := rabbit.Y + PLAYER_OFFSET_Y
+	rabbit.Sprite.Draw(x, y)
+
+	if rabbit.ShowItem {
```
Comment (JR-Mitchell) [inline cart/rabbit.go:83]: Is it worth adding extra methods so that we don't have too much nesting?

**Review by JR-Mitchell — CHANGES_REQUESTED** [review]

File: cart/rabbit.go [hunk 3]
```diff
@@ -30,14 +58,73 @@ func (rabbit *Rabbit) switchIdleFrame() {
 	rabbit.Sprite.Id = rabbit_main_frame + (4 * rabbit.Frame)
 }
 
+func (rabbit *Rabbit) drawThoughBubble(x, y, t int32) {
```
Comment (jwiggins) [inline cart/rabbit.go]: Should `drawThoughtBubble`

File: cart/rabbit.go [hunk 4]
```diff
@@ -30,14 +57,75 @@ func (rabbit *Rabbit) switchIdleFrame() {
...
+	x_item := x - RABBIT_ITEM_OFFSET
+	y_item := y - RABBIT_ITEM_OFFSET
+
+	if rabbit.BubbleState >= THOUGHT_BUBBLE_SMALL {
+		tic80.EllipseWithBorder(x_item+10, y_item+11, 2, 1, 12, 13)
+	}
+	if rabbit.BubbleState >= THOUGHT_BUBBLE_BIG {
+		tic80.EllipseWithBorder(x_item+3, y_item+3, 7, 6, 12, 13)
+		sprite.Draw(x_item, y_item)
+	}
```
Comment (jwiggins) [inline cart/rabbit.go:71]: So the overdraw is OK here?

**Review by JR-Mitchell — APPROVED** [review]
LGTM

**Review by jwiggins — APPROVED** [review]
🫠

**Pull request merged**